doc/tutorials/building_acrn_in_docker.rst

```diff
@@ -0,0 +1,158 @@
+
+.. code-block:: none
+
+   $ docker pull acrn/clearlinux:latest
+
+Build ACRN from Source in Docker
+********************************
+
+#. Run acrn/clearlinux:latest Docker image
+
+   .. code-block:: none
+
+      $ docker run -t -i --name acrn-hypervisor -w /workspace acrn/clearlinux:latest
+
+#. Clone the acrn-hypervisor repo
+
+   .. code-block:: none
+
+      $ git clone https://github.com/projectacrn/acrn-hypervisor
+
+#. Build acrn-hypervisor with default configuration
+
+   .. code-block:: none
+
+      # cd acrn-hypervisor
+      # make
+
+   After build is done, you could use the command ``exit`` to exit the Docker
+   container.
+
+#. Copy the artifacts from the Docker container to host
+
+   .. code-block:: none
+
+      $ docker cp acrn-hypervisor:/workspace/acrn-hypervisor/build .
+
+Build ACRN SOS Kernel in Docker
+*******************************
+
+#. Run acrn/clearlinux:latest Docker image
+
+   .. code-block:: none
+
+      $ docker run -t -i --name acrn-sos-kernel -w /workspace acrn/clearlinux:latest
+
+#. Clone the acrn-kernel repo
+
+   .. code-block:: none
+
+      # git clone https://github.com/projectacrn/acrn-kernel
+
+#. Build ACRN SOS kernel
+
+   .. code-block:: none
+
+      # cd acrn-kernel
+      # cp kernel_config_sos .config
+      # make olddefconfig
+      # make
+      # make modules_install INSTALL_MOD_PATH=out/
+
+   The last two commands build the bootable kernel image as ``arch/x86/boot/bzImage``,
+   and loadable kernel modules under the ``./out/`` folder.
+
+   After build is done, you could use the command ``exit`` to exit the Docker
+   container.
+
+#. Copy the artifacts from the Docker container to host
+
+   .. code-block:: none
+
+      $ docker cp acrn-sos-kernel:/workspace/acrn-kernel/arch/x86/boot/bzImage .
+      $ docker cp acrn-sos-kernel:/workspace/acrn-kernel/out/lib .
+
+Build ACRN UOS Preempt-rt Kernel in Docker
+******************************************
+
+#. Run acrn/clearlinux:latest Docker image
+
+   .. code-block:: none
+
+      $ docker run -t -i --name acrn-uos-preempt-rt -w /workspace acrn/clearlinux:latest
+
+#. Clone the ACRN preempt-rt kernel repo
+
+   .. code-block:: none
+
+      # git clone -b 4.19/preempt-rt https://github.com/projectacrn/acrn-kernel
+
+#. Build ACRN UOS preempt-rt kernel
+
+   .. code-block:: none
+
+      # cd acrn-kernel
+      # cp x86-64_defconfig .config
+      # make olddefconfig
+      # make
+      # make modules_install INSTALL_MOD_PATH=out/
+
+   The last two commands build the bootable kernel image as ``arch/x86/boot/bzImage``,
+   and loadable kernel modules under the ``./out/`` folder.
+
+   After build is done, you could use the command ``exit`` to exit the Docker
+   container.
+
+#. Copy the artifacts from the Docker container to host
+
+   .. code-block:: none
+
+      $ docker cp acrn-uos-preempt-rt:/workspace/acrn-kernel/arch/x86/boot/bzImage .
+      $ docker cp acrn-uos-preempt-rt:/workspace/acrn-kernel/out/lib .
```

Comment:
Can you add something that would be a natural next step for the developer to do after this?

Reply:
@dbkinder this doc just provide how to build using docker, about how to install the kernel, maybe use another doc to describe it, thanks